doc: Bluetooth: Mesh: Release notes for Bluetooth mesh (#4856)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-latest.rst

  * Renamed occurrences of Project CHIP to Matter.

Bluetooth Mesh
--------------

* Added:

  * Support for vendor specific mesh model :ref:`bt_mesh_silvair_enocean_srv_readme`.
  * API function to manually store pending RPL entries in the persistent storage without waiting for the timeout.

* Updated:

  * Samples are using :option:`CONFIG_NVS` instead of :option:`CONFIG_FCB` as default storage backend.
  * Renamed the following Kconfig options:

    * ``CONFIG_BT_MESH_LIGHT_CRTL_STORE_TIMEOUT`` renamed to :option:`CONFIG_BT_MESH_MODEL_SRV_STORE_TIMEOUT`, and default value is set to 0.

>   * The configuration setting for acknowledged message have been moved into Kconfig to make them public.

Reviewer: ```suggestion
  * The configuration settings for acknowledged messages have been moved into Kconfig to make them public.
```

  * Added persistent storage to the Scheduler model to restore previously configured entries on power up.

>   * Fixed several bugs.

Reviewer 2: Is there a reason why we don't want to be specific about the fixes? If we're talking about just a few bugs, we might as well list them?

Author: We should really lock through them then. And check if each one off them may have a impact on the user. Maybe we just should remove this line, and check if there is bug that is important for the user to know about.

  * Renamed occurrences of Project CHIP to Matter.

Bluetooth Mesh
--------------

* Added:

  * Support for vendor specific mesh model :ref:`bt_mesh_silvair_enocean_srv_readme`.
  * API function to manually store pending RPL entries in the persistent storage without waiting for the timeout.

> 

Reviewer 3: Please add (https://github.com/nrfconnect/sdk-nrf/pull/4918):
```
* A :member:`bt_mesh_scene_entry::recall_complete` callback that is called for each model that has a scene entry when recalling a scene data is done.

* Updated:

  * Fixed an issue by :ref:`bt_mesh_light_xyl_srv_readme` publishes its state values that was loaded from flash after power up.
  * Replaced the linked list of scene entries in the model contexts, with a lookup in ROM-allocated scene entries for each model in the composition data when storing and restoring.

>   * Updated so the transition pointer can be NULL, if no transition time parameters are provided in APIs.

Author: This will not be correct to write "to be", because it now can be NULL. Before it could not be NULL. Now it can be NULL, but it can be something else too.

  * Renamed Kconfig option ``CONFIG_BT_MESH_LIGHT_CTRL_STORE_TIMEOUT`` to :option:`CONFIG_BT_MESH_MODEL_SRV_STORE_TIMEOUT`, and default value is set to 0.

>   * Fixed an issue by added a timer that allows the :ref:`bt_mesh_light_ctrl_srv_readme` to resume operation after a certain delay.

Reviewer 2: Again, are we describing the solution to an issue that is not mentioned? And should we mention it, or are we ok with this?

Author: I think it sometime is better to describe what the solution is. I change this one any way.

>   * Updated the Light LC regulator to use instant transition time, to relived the application for overhead.

Reviewer 2: ```suggestion
  * Updated the Light LC regulator to use instant transition time to relieve the application from overhead.
```

  * Fixed an issue where undefined state for some sensor properties is valid states, and should be handled without giving errors.
  * Fixed an issue with store/recall of Light OnOff state in :ref:`bt_mesh_light_ctrl_srv_readme`.
  * Fixed an issue where :ref:`bt_mesh_lightness_srv_readme` publishes twice if extended by two models.

>   * Updated :ref:`bt_mesh_light_hue_srv_readme` and :ref:`bt_mesh_light_sat_srv_readme` to stored states upon a change.

Reviewer 2: ...to store their states upon a change? Or ...to stored states?

  * Fixed an issue where :ref:`bt_mesh_light_hue_srv_readme` and :ref:`bt_mesh_light_sat_srv_readme` did not erase model data on reset.
  * Fixed an issue where :ref:`bt_mesh_scene_srv_readme` called scene recall at startup.
  * Fixed an issue by publication of new value on scene recall in :ref:`bt_mesh_onoff_srv_readme` and :ref:`bt_mesh_lvl_srv_readme`.
  * Fixed issues where extended models did the store/recall instead of the extending model.
  * Updated the extending models by adding the extension API by default through Kconfig.
  * Force extension of :ref:`bt_mesh_lightness_srv_readme` to be initialized before :ref:`bt_mesh_light_ctrl_srv_readme`.

>   * Fixed an issue where :ref:`bt_mesh_light_ctrl_srv_readme` should disable control when the lightness is set by a message being received.

Reviewer 4: > * Fixed an issue where :ref:`bt_mesh_light_ctrl_srv_readme` should disable control when the lightness is set by receiving a message.

  * Added persistent storage to the :ref:`bt_mesh_scheduler_readme` to restore previously configured entries on power up.
  * Fixed an issue where CTL temperature bindings should use rounding operation for division in the binding formula.
  * Samples are using :option:`CONFIG_NVS` instead of :option:`CONFIG_FCB` as default storage backend.

>   * Fixed an issue in :ref:`bt_mesh_light_ctrl_srv_readme` by always set transition time to Fade Standby Manual state time when Light Off event is received.

Author: I would rather not, as it will be a lot of links. I think linking to the models is enough.

  * The configuration settings for acknowledged messages have been moved into Kconfig to make them public.
  * Fixed an issue where Occupancy On event did not transition to Fade On state even if Occupancy Mode is disabled.
  * Fixed an issue by adding a flag to :ref:`bt_mesh_onoff_srv_readme` to skip Default Transition Time on Generic OnOff Set (Unack) message.
  * Fixed an issue by correcting the bindings between Generic OnOff and Light OnOff.
  * Fixed an issue by clearing the internal sum in pi regulator when entering the OFF state of the state machine.
  * Fixed an issue where :ref:`bt_mesh_lightness_srv_readme` before it was stared could accidentally disabled :ref:`bt_mesh_light_ctrl_srv_readme`.
  * Fixed an issue by adding publication of Light Lightness Status message on boot up even if OnPowerUp state is OFF.
  * Fixed issues by adding publication of Light OnOff Status when disabling and on restore of Light LC.
  * Fixed an issue where the default values for temperature and range should be within valid range for the :ref:`bt_mesh_light_temp_srv_readme`.

>   * Removed sensor type ``BT_MESH_SENSOR_DELTA_DISABLED`` as it is removed from specification.

Reviewer 3: Please add (https://github.com/nrfconnect/sdk-nrf/pull/4743):
```
* `struct bt_mesh_model_ack_ctx` is replaced with :struct:`bt_mesh_msg_ack_ctx` from :ref:`zephyr:bluetooth_mesh_msg`.
```

  * Renamed occurrences of Project CHIP to Matter.

Bluetooth Mesh
--------------

* Added:

  * Support for vendor specific mesh model :ref:`bt_mesh_silvair_enocean_srv_readme`.

>   * API function to manually store pending RPL entries in the persistent storage without waiting for the timeout.

Author: This API is in zephyr so I am just adding the name.

* Updated:

>   * Fixed an issue by :ref:`bt_mesh_light_xyl_srv_readme` publishes its state values that was loaded from flash after power up.

Author: This flips the meaning to the opposite. We need a "did not" in there if we are going to write what was done before.

>   * Fixed an issue by :ref:`bt_mesh_light_xyl_srv_readme` publishes its state values that was loaded from flash after power up.

Reviewer 4: ```suggestion
  * Fixed an issue where :ref:`bt_mesh_light_xyl_srv_readme` did not publish the state values that were loaded from flash after powering up.
```

>   * Fixed an issue by :ref:`bt_mesh_light_xyl_srv_readme` publishes its state values that was loaded from flash after power up.

Author: Used Mia's suggestion instead.

  * Replaced the linked list of scene entries in the model contexts, with a lookup in ROM-allocated scene entries for each model in the composition data when storing and restoring.
  * Updated so the transition pointer can be NULL, if no transition time parameters are provided in APIs.
  * Renamed Kconfig option ``CONFIG_BT_MESH_LIGHT_CTRL_STORE_TIMEOUT`` to :option:`CONFIG_BT_MESH_MODEL_SRV_STORE_TIMEOUT`, and default value is set to 0.
  * Fixed an issue by added a timer that allows the :ref:`bt_mesh_light_ctrl_srv_readme` to resume operation after a certain delay.
  * Updated the Light LC regulator to use instant transition time, to relived the application for overhead.
  * Fixed an issue where undefined state for some sensor properties is valid states, and should be handled without giving errors.
  * Fixed an issue with store/recall of Light OnOff state in :ref:`bt_mesh_light_ctrl_srv_readme`.

>   * Fixed an issue where :ref:`bt_mesh_lightness_srv_readme` publishes twice if extended bye two models.

Reviewer 5: bye two models :)

  * Updated :ref:`bt_mesh_light_hue_srv_readme` and :ref:`bt_mesh_light_sat_srv_readme` to stored states upon a change.
  * Fixed an issue where :ref:`bt_mesh_light_hue_srv_readme` and :ref:`bt_mesh_light_sat_srv_readme` did not erase model data on reset.
  * Fixed an issue where :ref:`bt_mesh_scene_srv_readme` called scene recall at startup.
  * Fixed an issue by publication of new value on scene recall in :ref:`bt_mesh_onoff_srv_readme` and :ref:`bt_mesh_lvl_srv_readme`.
  * Fixed an issue where extended model did the store/recall instead of the extending model.
  * Updated the extending models by adding the extension API by default through Kconfig.
  * Force extension of :ref:`bt_mesh_lightness_srv_readme` to be initialized before :ref:`bt_mesh_light_ctrl_srv_readme`.
  * Fixed an issue where :ref:`bt_mesh_light_ctrl_srv_readme` should disable control when the lightness is set by a message being received.
  * Added persistent storage to the :ref:`bt_mesh_scheduler_readme` to restore previously configured entries on power up.
  * Fixed an issue where CTL temperature bindings should use rounding operation for division in the binding formula.

>   * Fixed an issue by not setting Generic OnOff in the :ref:`bt_mesh_ponoff_srv_readme` if it is extended by other model.

Reviewer 5: Isn't this the same 
```
Fixed an issue where extended model did the store/recall instead of the extending model.
```

  * Renamed occurrences of Project CHIP to Matter.

Bluetooth mesh
--------------

* Added:

  * Support for vendor-specific mesh model :ref:`bt_mesh_silvair_enocean_srv_readme`.
  * A new API function ``bt_mesh_rpl_pending_store`` to manually store pending RPL entries in the persistent storage without waiting for the timeout.
  * A :c:member:`bt_mesh_scene_entry::recall_complete` callback that is called for each model that has a scene entry when recalling a scene data is done.

* Updated:

>   * Updated the :ref:`bt_mesh_light_xyl_srv_readme` to publish its state values that was loaded from flash after powering up.

Reviewer 4: The word "was" is confusing or incorrect here. What was loaded from flash after powering up - the `bt_mesh_ligh_xyl_srv_readme` or the state values?

Here are two alternatives:

1) * Updated the :ref:`bt_mesh_light_xyl_srv_readme`  that was loaded from flash after powering up to publish its state values.
2) * Updated the :ref:`bt_mesh_light_xyl_srv_readme` to publish its state values that _were_ loaded from flash after powering up.

  * Replaced the linked list of scene entries in the model contexts, with a lookup in ROM-allocated scene entries.
  * Updated so the transition pointer can be NULL, if no transition time parameters are provided in APIs.
  * Renamed Kconfig option ``CONFIG_BT_MESH_LIGHT_CTRL_STORE_TIMEOUT`` to :option:`CONFIG_BT_MESH_MODEL_SRV_STORE_TIMEOUT`, and default value is set to 0.

>   * Adding a timer that allows the :ref:`bt_mesh_light_ctrl_srv_readme` to resume operation after a certain delay.

Reviewer 2: ```suggestion
  * Updated the :ref:`bt_mesh_light_ctrl_srv_readme` with a timer, allowing it to resume operation after a certain delay.
```

  * Samples are using :option:`CONFIG_NVS` instead of :option:`CONFIG_FCB` as the default storage backend.
  * Fixed an issue in :ref:`bt_mesh_light_ctrl_srv_readme` by always setting the transition time to a Fade Standby Manual state time when a Light Off event is received.
  * Fixed an issue by reporting maximum remaining time for all components for CTL state transition time when GET is processed.
  * Fixed an issue where a deleted :ref:`bt_mesh_scene_srv_readme` did not delete all its pages from the file system.
  * Fixed an issue where Sensor Threshold was trimmed and an invalid value was calculated.
  * Updated :ref:`bt_mesh_scheduler_srv_readme` to no longer extend :ref:`bt_mesh_scene_srv_readme`, but they must be present on the same element.
  * Moved the configuration settings for acknowledged messages into Kconfig to make them public.
  * Fixed an issue where an Occupancy On event did not transition to a Fade On state even if Occupancy Mode is disabled.
  * Added a flag to :ref:`bt_mesh_onoff_srv_readme` to skip Default Transition Time on Generic OnOff Set (Unack) message.

>   * Fixed an issue by correcting the bindings between Generic OnOff state and Light OnOff state.

Reviewer 4: ```suggestion
  * Fixed an issue by correcting the bindings between the Generic OnOff state and the Light OnOff state.
```

  * Fixed an issue by clearing the internal sum in the proportional-integral (PI) feedback regulator when entering the OFF state of the state machine.

>   * Fixed an issue where :ref:`bt_mesh_lightness_srv_readme` could accidentally disable :ref:`bt_mesh_light_ctrl_srv_readme` before it was stared.

Reviewer 4: ```suggestion
  * Fixed an issue where :ref:`bt_mesh_lightness_srv_readme` could accidentally disable :ref:`bt_mesh_light_ctrl_srv_readme` before it was started.
```

>   * Fixed an issue by adding publication of the Light Lightness Status message on startup even if OnPowerUp state is OFF.

Reviewer 4: ```suggestion
  * Fixed an issue by publishing the Light Lightness Status message on startup even if the OnPowerUp state is OFF.
```

>   * Fixed issues by adding publication of the Light OnOff Status when disabling and restoring the Light LC state.

Reviewer 4: ```suggestion
  * Fixed issues by publishing the Light OnOff Status when disabling and restoring the Light LC state.
```

  * Fixed an issue where temperature and range should be within a valid default range for the :ref:`bt_mesh_light_temp_srv_readme`.
  * Removed sensor type ``BT_MESH_SENSOR_DELTA_DISABLED`` as it is removed from specification.

>   * Replaced ``struct bt_mesh_model_ack_ctx`` with :struct:`bt_mesh_msg_ack_ctx` from :ref:`zephyr:bluetooth_mesh_msg`.

Reviewer 6: ```suggestion
  * Replaced ``struct bt_mesh_model_ack_ctx`` with `` struct bt_mesh_msg_ack_ctx` from :ref:`zephyr:bluetooth_mesh_msg`.
```

to resolve build error

Reviewer 2: I totally missed this one!
